csharp/ql/lib/semmle/code/csharp/security/dataflow/UrlRedirectQuery.qll

```diff
@@ -184,8 +184,12 @@ private class InterpolationSanitizer extends Sanitizer {
  */
 private class StringFormatSanitizer extends Sanitizer {
   StringFormatSanitizer() {
-    exists(FormatCall c, Expr e | c = this.getExpr() and e = c.getFormatExpr() |
-      e.(StringLiteral).getValue().splitAt("{0}", 0).matches("%?%")
+    exists(FormatCall c, Expr e, int index, string format |
+      c = this.getExpr() and e = c.getFormatExpr()
+    |
+      format = e.(StringLiteral).getValue() and
```

Comment:
No, it won't catch cases like that.
The same is the case for the original `ConcatenationSanitizer`.
The following would lead to a false positive:
```
var left = "foo.asp?param=";
ctx.Response.Redirect(left + url);
```
I think the "preferred" sanitization method is doing url encoding of the input and these concat like sanitizers are here for cases where the created string is safe (by a very local inspection).